File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        index = -1
        for i in range(len(haystack) - len(needle) + 1):
            if haystack[i:i+len(needle)] == needle:
                index = i
                break
        return index


if __name__ == '__main__':
    solution = Solution()
    print(solution.strStr("sadbut", "sad"))
    print(solution.strStr("sabbutsad", "sad"))
    print(solution.strStr("leetcode", "leeto"))

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.go
package main

//func strStr(haystack string, needle string) int {
//	return strings.Index(haystack, needle)
//}

// 滑动窗口
func strStr(haystack string, needle string) int {
	l := -1
	for i := 0; i <= len(haystack)-len(needle); i++ {
		if haystack[i:i+len(needle)] == needle {
			l = i
			break
		}
	}
	return l
}

func main() {
	print(strStr("sadbut", "sad"))
}
